File: swimming-contest-software/src/main/java/swimmingApp/domain/dtos/SwimmerDTO.java
package swimmingApp.domain.dtos;

import swimmingApp.domain.entities.Swimmer;

import java.util.List;

public class SwimmerDTO {
    private Swimmer swimmer;
    private List<RaceDetailsDTO> raceDetailsDTOS;

    public SwimmerDTO(Swimmer swimmer, List<RaceDetailsDTO> races) {
        this.swimmer = swimmer;
        this.raceDetailsDTOS = races;
    }

    public Swimmer getSwimmer() {
        return swimmer;
    }

    public List<RaceDetailsDTO> getRaceDetailsDTOS() {
        return raceDetailsDTOS;
    }
}

File: MPP-lab-project/src/Domain/DTOs/SwimmerDTO.cs
using System.Collections.Generic;
using System.Linq;

namespace MPP_lab_project.Domain.DTOs;

public class SwimmerDTO
{
    private readonly Swimmer _swimmer;
    private readonly List<RaceDetailsDTO> _raceDetailsDTOs;

    public string FirstName => _swimmer.FirstName;

    public string LastName => _swimmer.LastName;

    public int Age => _swimmer.Age;

    public string Races
    {
        get
        {
            return _raceDetailsDTOs.Aggregate("", (subtotal, element) => subtotal + element);
        }
    }

    public SwimmerDTO(Swimmer swimmer, List<RaceDetailsDTO> raceDetailsDtOs)
    {
        _swimmer = swimmer;
        _raceDetailsDTOs = raceDetailsDtOs;
    }
}